system-x/services/splunk/src/main/java/software/tnb/splunk/resource/openshift/OpenshiftSplunk.java

```diff
@@ -200,14 +240,18 @@ private CustomResourceDefinitionContext createSplunkContext() {
                 .withGroup(crd.getSpec().getGroup())
                 .withPlural(crd.getSpec().getNames().getPlural())
                 .withScope(crd.getSpec().getScope())
-                .withVersion("v3");
+                .withVersion(CRD_API);
             crdContext = builder.build();
         }
         return crdContext;
     }
 
     @Override
     public int apiPort() {
-        return API_PORT;
+        if (getConfiguration().getProtocol().equals(SplunkProtocol.HTTPS)) {
```

Comment:
can be simplified to `return getConfiguration().getProtocol() == SplunkProtocol.HTTPS ? 443 : 80`